ouroboros-consensus-protocol/src/ouroboros-consensus-protocol/Ouroboros/Consensus/Protocol/Praos.hs

```diff
@@ -652,10 +652,12 @@ doValidateKESSignature praosMaxKESEvo praosSlotsPerKESPeriod stakeDistribution o
 
   currentIssueNo :: Maybe Word64
   currentIssueNo
-    | Map.member hk ocertCounters = Map.lookup hk ocertCounters
-    | Set.member (coerceKeyRole hk) (Map.keysSet stakeDistribution) =
-        Just 0
-    | otherwise = Nothing
+    | r@Just {} <- Map.lookup hk ocertCounters
+      = r
+    | Set.member (coerceKeyRole hk) (Map.keysSet stakeDistribution)
```

Comment:
Just in case you feel like it because you already are in the area; also happy to merge what you have:
```suggestion
  | Map.member (coerceKeyRole hk) stakeDistribution
```
Could also use the `Alternative` instance of `Maybe` :thinking:

Reply:
:thinking: `Alternative` - how?

Reply:
Ie like this:
```haskell
  currentIssueNo =
    Map.lookup hk ocertCounters
      <|> (0 <$ Map.lookup (coerceKeyRole hk) stakeDistribution)
```

Reply:
I agree, it's a bit too far.